README.md

```diff
@@ -10,35 +10,75 @@ This library uses a fluent API that is separated in 3 parts:
-* If the HTTP response has a *401* status code, we return `EitherStrict.Right(OptionStrict.Nothing)`
-* If the HTTP response has a *2XX* status code, we return `EitherStrict.Right(OptionStrict.Just(operationResult))`, where `operationResult` is the data typed obtained by deserializing the JSON response body.
+![restclient example 1](docs/img/restclient_cat_example_post.png?raw=true)
 
-No matter what the API responds with, we return the user a type specifying the information he wants from the API, mapping the HTTP response to a C# type (that is composed of various types like `EitherStrict`, `OptionStrict`, etc).
+We can also do the usual PUT and DELETE calls:
 
-### Processors
+![restclient example 1](docs/img/restclient_cat_example_put.png?raw=true)
 
-To achieve this, rest-client uses a pipeline of processors that check whether they find a specific HTTP status code, and if they find it they return a value of the returning type.
+![restclient example 1](docs/img/restclient_cat_example_delete.png?raw=true)
 
-* `EitherRestErrorProcessor` returns values of type `EitherStrict<RestBusinessError, TResult>`. It checks if the response has an unsuccessful status code (one different than *2XX*). If it does, then it returns a `RestBusinessError`; if it doesn't, then it delegates the processing to the rest of the pipeline so it returns a `TResult` value.
+In the case of `DeleteCat`, we also need to pass `Unit.Default` to indicate that we don't want anything passed in the request body. If we needed to call DELETE passing a JSON object in the request body, we'd replace `Unit.Default` with the object to serialize.
+
+## Managing 404 status codes
+
+For a more complex example, imagine "GET http://api.catsforall.com/cats/myCatName" returns a _404 Not Found_ status code if the cat doesn't exist. With our current API, in that scenario we throw an exception. However, what if we want to handle that case programatically? We can do that in the following way:
+
+![restclient example 1](docs/img/restclient_cat_example_get_option.png?raw=true)
+
+The only thing that changed from the original method is the return type, which changes from `Cat` to `OptionStrict<Cat>`, and we added a line `AddProcessors(new OptionAsNotFoundProcessor<Cat>())`. 
+
+Let's talk about the change in the returning type first. `OptionStrict<T>` is a type from the [Csharp-Monad](https://github.com/louthy/csharp-monad) library, it represents a nullable type (which can be either a value type or reference type). A value with type `OptionStrict<T>` is either empty or has a value. We can use this type to represent our new method: If the API call returns a 200 status code with the JSON body, then we return a `Cat` value; however, if the API call returns a 404 status code, we return an empty value (representd by `Option<Cat>.Nothing`).
```

Comment:
which I probably resume (advantages over null comparison) before showing the code that uses the OptionAsNotFoundProcessor